diff -uNr DPO_RT5572_LinuxSTA_2.6.1.3_20121022/Makefile DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/Makefile
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022/Makefile	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/Makefile	2014-02-20 03:58:49.564318455 +0100
@@ -27,7 +27,7 @@
 RTMP_SRC_DIR = $(RT28xx_DIR)/RT$(MODULE)
 
 #PLATFORM: Target platform
-PLATFORM = PC
+#PLATFORM = PC
 #PLATFORM = 5VT
 #PLATFORM = IKANOS_V160
 #PLATFORM = IKANOS_V180
@@ -63,6 +63,7 @@
 #PLATFORM = UBICOM_IPX8
 #PLATFORM = INTELP6
 #PLATFORM = MSTAR
+PLATFORM = CNS21XX
 
 #APSOC
 ifeq ($(MODULE),3050)
@@ -208,6 +209,11 @@
 CROSS_COMPILE = arm-linux-
 endif
 
+ifeq ($(PLATFORM),CNS21XX)
+#LINUX_SRC = /project/stable/Gmtek/snapgear-uclibc/linux-2.6.x
+CROSS_COMPILE = arm-linux-
+endif
+
 ifeq ($(PLATFORM),INF_TWINPASS)
 # Linux 2.6
 #LINUX_SRC = /lib/modules/$(shell uname -r)/build
@@ -401,29 +407,29 @@
 	$(MAKE) -C $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules
 endif
 
-ifeq ($(RT28xx_MODE),AP)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.ko /tftpboot
-endif
-	rm -f os/linux/rt$(MODULE)ap.ko.lzma
-	/root/bin/lzma e os/linux/rt$(MODULE)ap.ko os/linux/rt$(MODULE)ap.ko.lzma
-else	
-ifeq ($(RT28xx_MODE),APSTA)
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.ko /tftpboot
-endif
-else
-	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.ko /tftpboot
-ifeq ($(OSABL),YES)
-	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.ko /tftpboot
-	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.ko /tftpboot
-endif
-endif
-endif
+# ifeq ($(RT28xx_MODE),AP)
+# 	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)ap.ko /tftpboot
+# ifeq ($(OSABL),YES)
+# 	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)ap.ko /tftpboot
+# 	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)ap.ko /tftpboot
+# endif
+# 	rm -f os/linux/rt$(MODULE)ap.ko.lzma
+# 	/root/bin/lzma e os/linux/rt$(MODULE)ap.ko os/linux/rt$(MODULE)ap.ko.lzma
+# else	
+# ifeq ($(RT28xx_MODE),APSTA)
+# 	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)apsta.ko /tftpboot
+# ifeq ($(OSABL),YES)
+# 	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)apsta.ko /tftpboot
+# 	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)apsta.ko /tftpboot
+# endif
+# else
+# 	cp -f $(RT28xx_DIR)/os/linux/rt$(MODULE)sta.ko /tftpboot
+# ifeq ($(OSABL),YES)
+# 	cp -f $(RT28xx_DIR)/os/linux/rtutil$(MODULE)sta.ko /tftpboot
+# 	cp -f $(RT28xx_DIR)/os/linux/rtnet$(MODULE)sta.ko /tftpboot
+# endif
+# endif
+# endif
 endif
 
 
diff -uNr DPO_RT5572_LinuxSTA_2.6.1.3_20121022/include/mlme.h DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/include/mlme.h
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022/include/mlme.h	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/include/mlme.h	2014-02-08 19:19:25.050880665 +0100
@@ -129,9 +129,9 @@
 #define BSS_NOT_FOUND                    0xFFFFFFFF
 
 
-#ifdef CONFIG_STA_SUPPORT
+//#ifdef CONFIG_STA_SUPPORT
 #define MAX_LEN_OF_MLME_QUEUE            40 /*10 */
-#endif /* CONFIG_STA_SUPPORT */
+//#endif /* CONFIG_STA_SUPPORT */
 
 #define SCAN_PASSIVE                     18		/* scan with no probe request, only wait beacon and probe response */
 #define SCAN_ACTIVE                      19		/* scan with probe request, and wait beacon and probe response */
diff -uNr DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/config.mk DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/os/linux/config.mk
--- DPO_RT5572_LinuxSTA_2.6.1.3_20121022/os/linux/config.mk	2012-10-22 05:52:59.000000000 +0200
+++ DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/os/linux/config.mk	2014-02-08 19:12:27.626895625 +0100
@@ -23,7 +23,7 @@
 
 # Support Wpa_Supplicant
 # i.e. wpa_supplicant -Dralink
-HAS_WPA_SUPPLICANT=n
+HAS_WPA_SUPPLICANT=y
 
 
 # Support Native WpaSupplicant for Network Maganger
@@ -32,7 +32,7 @@
 # what if user want to use wpa_supplicant to serve P2P function/feature, 
 # in case, it must use Ralink Propriectary wpa_supplicant to do.
 # and this compile flag will report P2P Related Event to Ralink wpa_supplicant.
-HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=n
+HAS_NATIVE_WPA_SUPPLICANT_SUPPORT=y
 
 #Support Net interface block while Tx-Sw queue full
 HAS_BLOCK_NET_IF=n
@@ -975,6 +975,12 @@
 	export CFLAGS        
 endif
 
+ifeq ($(PLATFORM),CNS21XX)
+	CFLAGS := -v -D__KERNEL__ -DMODULE -I$(LINUX_SRC)/include -I$(RT28xx_DIR)/include -Wall -Wstrict-prototypes -Wno-trigraphs -O2 -fno-strict-aliasing -fno-common -Uarm -fno-common -pipe -march=armv4 -malignment-traps -msoft-float $(WFLAGS)
+        EXTRA_CFLAGS := -v $(WFLAGS) -I$(RT28xx_DIR)/include
+	export CFLAGS
+endif
+
 ifeq ($(PLATFORM),SMDK)
         EXTRA_CFLAGS := $(WFLAGS) -I$(RT28xx_DIR)/include
 endif
Binary files DPO_RT5572_LinuxSTA_2.6.1.3_20121022/tools/bin2h and DPO_RT5572_LinuxSTA_2.6.1.3_20121022-snake/tools/bin2h differ
